lxd/instance/drivers/driver_qemu.go

```diff
@@ -5778,11 +5795,12 @@ func (d *qemu) writeInstanceData() error {
 	}
 
 	out, err := json.Marshal(struct {
-		Name     string                         `json:"name"`
-		Location string                         `json:"location"`
-		Config   map[string]string              `json:"config,omitempty"`
-		Devices  map[string]deviceConfig.Device `json:"devices,omitempty"`
-	}{d.Name(), location, userConfig, d.expandedDevices})
+		Name        string                         `json:"name"`
```

Comment:
@stgraber can we use a shared struct type here to link this with `type instanceData struct` in lxd-agent, as would also avoid them diverging (as in typo previously).

Reply:
Could put it in `"github.com/lxc/lxd/lxd/device/config"` like `NICConfig` which is used for a similar thing (communicating config to lxd-agent).

Reply:
Doesn't make sense to put this in `lxd/device` and not sure what other package we can currently put this in which won't pull in cgo related things.

Reply:
`lxd/instance/instancetype` also has some lxd-agent related structs in.

Reply:
I'd argue that `VMAgentMount` should be moved into `device/config`, but thats for another day :)

Reply:
Added a commit